### PR TITLE
organize navbar in overview->details order

### PR DESCRIPTION
**Pull request by story645**
Um this is a first pass rather than something I'm strongly attached to, but I'm proposing the following story right to left because it goes from overview of the library to details on demand. I continue to not really know where to place tutorials but the middle kinda makes sense

- plot types -> user guide -> tutorials -> examples -> reference

### CODE REVIEW

**Review by tacaswell — APPROVED** [review]

File: mpl_sphinx_theme/mpl_nav_bar.html [hunk 1]
```diff
@@ -3,16 +3,16 @@
...
   <li class="nav-item">
     <a class="reference internal nav-link" href="{{ mpl_path('tutorials/index') }}">Tutorials</a>
   </li>
   <li class="nav-item">
-    <a class="reference internal nav-link" href="{{ mpl_path('api/index') }}">Reference</a>
+    <a class="reference internal nav-link" href="{{ mpl_path('gallery/index') }}">Examples</a>
   </li>
-  <li class="nav-item">
-    <a class="reference internal nav-link" href="{{ mpl_path('users/index') }}">User guide</a>
+   <li class="nav-item">
```
Comment (QuLogic) [inline mpl_sphinx_theme/mpl_nav_bar.html:14]: Extra space
```suggestion
  <li class="nav-item">
```

**Comment by timhoffm** [comment]
Ok by me. But note that the User Guide page cuurently is more a global TOC and contains the others. By content I'd does not fit in an overview -> detail order. But that's a more general issue of our general sphinxpage hierarchy..

**Comment by timhoffm** [comment]
> Agreeish but @jklymak especially has been putting a lot of work into making it more overviewish.

I know and appreciate that. We've had detailed discussions. At some point we decided to merge the current state. It's already much better than before. I'm just reminding that there is still work to be done. And the intended ordering of this PR is not yet reflected by the content behind the menu title "User guide".

**Review by QuLogic — APPROVED** [review]
It looks like the User guide does seem overview-y now, so this seems good.

**Pull request merged**